### PR TITLE
Added all available and relevant producer id's for ATI custom variable

### LINKED ISSUE #2839: Update ATI (analytics) for Arabic, Urdu, Pashto articles (and producers for all WS services)
**Is your feature request related to a problem? Please describe.**
We should ensure that the WS Languages stats destination values are used for World Service sites.

We already did this for Persian in https://github.com/bbc/simorgh/issues/1699 and so the requirement may already be covered by that work - this issue is to test this and do any further dev work that may be required (which may be none).

There is a Chrome plugin called ATI Tag Inspector that can be used to verify the values we are sending. An ATI login is also required - please speak to @jimjohnsonrollings or @lizcameron for more information.

We should also ensure we pass in all of these fields for Arabic:

Field | Value
-- | --
Destination (s) | WS_NEWS_LANGUAGES: 598342 (live), 598343 (test)
Producer | 5
Chapter | Null
Page (p) | arabic.articles.c0000000000o.page
appName (x3) | news-arabic

These for Urdu:

Field | Value
-- | --
Destination (s) | WS_NEWS_LANGUAGES: 598342 (live), 598343 (test)
Producer | 95
Chapter | Null
Page (p) | urdu.articles.c0000000000o.page
appName (x3) | news-urdu

These for Pashto:

Field | Value
-- | --
Destination (s) | WS_NEWS_LANGUAGES: 598342 (live), 598343 (test)
Producer | 68
Chapter | Null
Page (p) | pashto.articles.c0000000000o.page
appName (x3) | news-pashto

**Describe the solution you'd like**

**Describe alternatives you've considered**
N/A

**Testing notes**
Testing required here will be making sure that the responses found within the ATI terminal (Chrome Plugin) are as expected for both AMP & Canonical.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Partly solves #2839
Solution has been split with #2995

**Overall change:** Added all producer Id's that we have for most of our world services for ATI analytics as suggested by @sareh.

**Code changes:**

- Added ID's based on the routing documentation on the Simorgh [wiki](https://github.com/bbc/simorgh/wiki/Simorgh-Routing) which I cross referenced from our echo docs ATI reporting section [here](http://bbc.github.io/echo-docs/documentation/reporting-ati/#valid-producer-values)
- There weren't any ATI producer codes for the following regions however (EDIT: These have now been added after Jim's [comment](https://github.com/bbc/simorgh/pull/2991#issuecomment-519479492)):
**Naidheachdan (Scottish Gaelic)
Zhongwen (Simplified Chinese)**

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by jimjohnsonrollings** [comment]
Zhongwen should be 38 ('Chinese').
Naidheachdan should be 79 ('Scotland') - but note that Naidheachdan should be a different `s` (aka Destination) as well - 598291 (live) and 598292 (test).
Cymru Fyw should be 100 ('Wales') and the same Destinations as Naidheachdan.

**Comment by sareh** [comment]
@jimjohnsonrollings
Do Cymru Fyw and Naidheachdan have the same PS/GNL split of destination buckets like News does?
Just wondering, since I believe they are served over both '.co.uk' and '.com'.
https://github.com/bbc/simorgh/blob/bc28bb267d06b839b17aefada7c5ea7e1dcb26b2/src/app/lib/analyticsUtils/index.js#L5

**Comment by jimjohnsonrollings** [comment]
Oh yes. Good spot. I’ll look up the GNL destinations.  Also this reminds me that Japanese should go in the GNL destination with them instead of WS.

**Comment by jimjohnsonrollings** [comment]
That comment via email did not work as well as I’d hoped :(

**Comment by jimjohnsonrollings** [comment]
So:
GNL_NEWS_LANGUAGES = 598289
GNL_NEWS_LANGUAGES_TEST = 598290

These should be used for:
- Naidheachdan users who are outside the UK
- Cymru Fyw users who are outside the UK
- Japanese (all users)

**Comment by HarveyPeachey** [comment]
> So:
> GNL_NEWS_LANGUAGES = 598289
> GNL_NEWS_LANGUAGES_TEST = 598290
> 
> These should be used for:
> 
> * Naidheachdan users who are outside the UK
> * Cymru Fyw users who are outside the UK
> * Japanese (all users)

In the logic for working out what destination is chosen based on the location where the user is visiting from it has this: 
`  if (service === 'news') {
    destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';`
It can be found in this [file](https://github.com/bbc/simorgh/blob/latest/src/app/contexts/RequestContext/getStatsDestination/index.js)

Is the logic correct because it's appending "NEWS_PS" and "NEWS_GNL" but as you've stated it's "GNL_NEWS" is there a difference between them in regards to ATI?

**Comment by jimjohnsonrollings** [comment]
Unfortunately all the systems have different conventions for the order of the naming elements. Ultimately the number is what’s important.

File: src/app/lib/analyticsUtils/index.js [hunk 1]
```diff
@@ -32,12 +32,51 @@ export const isLocServeCookieSet = platform => {
 
 export const getProducer = service => {
   const producers = {
```
Comment (12) [inline src/app/lib/analyticsUtils/index.js]: I think these would be better homed in the [service configs](https://github.com/bbc/simorgh/tree/latest/src/app/lib/config/services)

**Review by 12 — CHANGES_REQUESTED** [review]
Looking good! Got a suggestion though to help keep service-related things all in one place 🙂

File: src/app/lib/analyticsUtils/index.js (hunk 1 above)
Comment (jamesdonoh) [reply src/app/lib/analyticsUtils/index.js]: Good idea!

Comment (jimjohnsonrollings) [reply src/app/lib/analyticsUtils/index.js]: In future, News won't have a 1:1 service:producer relationship.

File: src/app/containers/MediaPlayer/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -51,4 +51,4 @@ exports[`MediaPlayer Calls the correct props Calls the canonical player when pla
 
 exports[`MediaPlayer Fails and returns early when component is toggled off 1`] = `null`;
 
-exports[`MediaPlayer Fails and returns early when there is no versionId 1`] = `null`;
+exports[`MediaPlayer Fails and returns early when there is no versionId 1`] = `null`;
```
Comment (12) [inline src/app/containers/MediaPlayer/__snapshots__/index.test.jsx.snap]: I don't think this file needs committing.

`git checkout latest -- src/app/containers/MediaPlayer/__snapshots__/index.test.jsx.snap` will reset the file to latest

**Review by 12 — CHANGES_REQUESTED** [review]
Checking this out locally, I can see that on frontpages, the producer value is no longer being sent (it currently is being sent on test).

Aside from that, this is looking good 👍 

image

**Comment by HarveyPeachey** [comment]
> Checking this out locally, I can see that on frontpages, the producer value is no longer being sent (it currently is being sent on test).
> 
> Aside from that, this is looking good 👍
> 
> image

Added the functionality for front pages and media as well. Tested it and it seems to be working fine:

**Front Page**
Screenshot 2019-08-15 at 15 05 09

**Media Page**
Screenshot 2019-08-15 at 15 04 27

**Review by 12 — APPROVED** [review]
good work 👍

**Review by RayNjeri — APPROVED** [review]

**Comment by sareh** [comment]
Great work, Harvey!
Since the getDestination method has not been updated in this PR, could you please open a separate PR to add these?
https://github.com/bbc/simorgh/pull/2991#issuecomment-519825139

**Review by sareh — APPROVED** [review]

**Comment by HarveyPeachey** [comment]
> Great work, Harvey!
> Since the getDestination method has not been updated in this PR, could you please open a separate PR to add these?
> [#2991 (comment)](https://github.com/bbc/simorgh/pull/2991#issuecomment-519825139)

Thanks Sareh, I have a PR open already: #3249 

The code's done I'm just making some unit tests for it.

**Review by ryanmccombe — APPROVED** [review]

**Pull request merged**